platforms/ios/demo/src/MapViewController.m

```diff
@@ -199,11 +211,38 @@ - (void)viewDidLoad
 
     self.mapViewDelegate = [[MapViewControllerDelegate alloc] init];
     self.gestureDelegate = [[MapViewControllerRecognizerDelegate alloc] init];
+    self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.delegate = self;
+    // Enable for Location Tracking
+    // [self.locationManager requestAlwaysAuthorization];
 }
 
-- (void)didReceiveMemoryWarning
-{
-    [super didReceiveMemoryWarning];
+- (void)beginBackgroundLocationTracking {
+    self.locationManager.activityType = CLActivityTypeOther;
+    self.locationManager.desiredAccuracy = kCLLocationAccuracyBest;
+    self.locationManager.pausesLocationUpdatesAutomatically = NO;
+    self.locationManager.allowsBackgroundLocationUpdates = YES;
+    [self.locationManager startUpdatingLocation];
+    [self.locationManager startUpdatingHeading];
+}
+
+#pragma mark - Location Manager Delegate
+
+- (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray<CLLocation *> *)locations {
+    NSLog(@"Locations came in - %@", locations);
+    TGGeoPoint point = TGGeoPointMake(locations[0].coordinate.longitude, locations[0].coordinate.latitude);
+    [self.locationTrackingMarker pointEased:point seconds:1.0 easeType:TGEaseTypeCubic];
+}
+
+- (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status {
+    switch (status) {
+        case kCLAuthorizationStatusAuthorizedAlways:
+            // Enable to turn on background location tracking - be aware we don't shut this off ever, so the sample app will consume battery life if allowed to run in the background.
+            // [self beginBackgroundLocationTracking];
```

Comment:
Not sure if this is expected but if I uncomment this I get the following:

```
2017-10-31 11:28:48.201296-0400 tangram[5255:87931] *** Assertion failure in -[CLLocationManager setAllowsBackgroundLocationUpdates:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/CoreLocationFramework_Sim/CoreLocation-2235.0.2.2/Framework/CoreLocation/CLLocationManager.m:626
2017-10-31 11:28:48.203696-0400 tangram[5255:87931] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid parameter not satisfying: !stayUp || CLClientIsBackgroundable(internal->fClient)'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010631f1cb __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x0000000109b4ef41 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000106324362 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x00000001095f3089 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 193
	4   CoreLocation                        0x000000010677eb67 CLClientRetrieveData + 28048
	5   tangram                             0x00000001052f9269 -[MapViewController beginBackgroundLocationTracking] + 185
	6   tangram                             0x00000001052f933e -[MapViewController locationManager:didChangeAuthorizationStatus:] + 62
	7   CoreLocation                        0x00000001067892a0 CLClientRetrieveData + 70857
	8   CoreLocation                        0x0000000106788f64 CLClientRetrieveData + 70029
	9   CoreLocation                        0x00000001067731f3 CLClientInvalidate + 1048
	10  CoreFoundation                      0x00000001062c220c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	11  CoreFoundation                      0x00000001062a6a3b __CFRunLoopDoBlocks + 203
	12  CoreFoundation                      0x00000001062a683b __CFRunLoopRun + 2875
	13  CoreFoundation                      0x00000001062a5a89 CFRunLoopRunSpecific + 409
	14  GraphicsServices                    0x000000010d8d49c6 GSEventRunModal + 62
	15  UIKit                               0x0000000106ad9d30 UIApplicationMain + 159
	16  tangram                             0x00000001052f9523 main + 99
	17  libdyld.dylib                       0x000000010a70ed81 start + 1
)
```

Reply:
Interesting! I'll take a look.

Reply:
Ah-ha yeah somehow the `UIBackgroundModes` key got omitted from the Info.plist when I uploaded this. Fixed now.

Reply:
Same happens with latest version.